libqpdf/QPDFWriter.cc

```diff
@@ -1750,7 +1759,7 @@ QPDFWriter::writeObjectStream(QPDFObjectHandle object)
     adjustAESStreamLength(length);
     writeString(" /Length " + std::to_string(length));
     writeStringQDF("\n ");
-    if (compressed) {
+    if (compressed && length > 0) {
```

Comment:
Again, just set compressed to false if the object stream is empty (i.e. if m->object_stream_to_objects[old_id] is empty).

Reply:
Disabling compression in `QPDFWriter::willFilterStream()` is good enough in this case, so I just dropped this change.